storage/2020-08-04/file/files/range_put_file.go

```diff
@@ -17,6 +17,10 @@ func (c Client) PutFile(ctx context.Context, shareName, path, fileName string, f
 		return fmt.Errorf("error loading file info: %s", err)
 	}
 
+	if fileInfo.Size() == 0 {
+		return fmt.Errorf("file is empty which is not supported")
+	}
```

Comment:
could we also apply this fix to `2023-11-03` too?

Reply:
Actually I'm going to pull this out into another PR since this is otherwise good to go 👍